lib/ansible/utils/helpers.py

```diff
@@ -41,3 +43,19 @@ def object_to_dict(obj, exclude=None):
     if exclude is None or not isinstance(exclude, list):
         exclude = []
     return dict((key, getattr(obj, key)) for key in dir(obj) if not (key.startswith('_') or key in exclude))
+
+
+def update_file_if_different(filename, data):
```

Comment:
This doesn't really belong here as nothing in ansible's runtime needs it.  Unfortunately, we don't have a directory where it would be picked up by the build scripts.  So maybe a separate, private file in here like lib/ansible/utils/_build_helpers.py.